graylog2-web-interface/src/components/common/InteractableModal.jsx

```diff
@@ -90,56 +90,56 @@ const InteractableModal = ({
   const [resizeSize, setResizeSize] = useState(size);
 
   const handleDragStop = (event, newPosition) => {
-    const { x, y, node } = newPosition;
-    const width = parseFloat(node.style.width);
-    const height = parseFloat(node.style.height);
-    const bodyWidth = document.body.offsetWidth;
-    const bodyHeight = document.body.offsetHeight;
+    const { x, y } = newPosition;
+
+    const setPosition = { x, y };
+    setDragPosition(setPosition);
+    onDrag(setPosition);
+  };
+
+  const handleBrowserResize = debounce(() => {
+    const { x, y } = dragPosition;
+    const { width, height } = resizeSize;
+    const { innerWidth, innerHeight } = window;
     const boundingBox = {
       top: 0,
-      right: bodyWidth - width,
-      bottom: bodyHeight - height,
+      right: innerWidth - width,
+      bottom: innerHeight - height,
       left: 0,
     };
 
     const tooFarLeft = x < boundingBox.left;
     const tooFarRight = x > boundingBox.right;
-    const newRight = tooFarRight ? bodyWidth - width : x;
-    const newX = tooFarLeft ? 0 : newRight;
+    const newRight = tooFarRight ? boundingBox.right : x;
+    const nextX = tooFarLeft ? 0 : newRight;
 
     const tooFarUp = y < boundingBox.top;
     const tooFarDown = y > boundingBox.bottom;
-    const newDown = tooFarDown ? bodyHeight - height : y;
-    const newY = tooFarUp ? 0 : newDown;
-
-    const setPosition = {
-      x: newX < 0 ? 0 : newX,
-      y: newY < 0 ? 0 : newY,
-    };
-    setDragPosition(setPosition);
-    onDrag(setPosition);
-  };
+    const newDown = tooFarDown ? boundingBox.bottom : y;
+    const nextY = tooFarUp ? 0 : newDown;
 
-  const handleBrowserResize = debounce((event) => {
-    handleDragStop(event, { node: { style: { ...resizeSize } }, ...dragPosition });
+    handleDragStop(null, { x: nextX, y: nextY });
   }, 150);
 
   useEffect(() => {
     setDragHandleClassName(dragHandleRef.current.classList[0]);
+  }, []);
+
+  useEffect(() => {
```

Comment:
Might not technically need two `useEffects`. I don't think re-running `setDragHandleClassName` will cause much performance issue. Not sure which direction you think makes the most sense

Reply:
For me it makes it clearer having both in different `useEffect`. Also that way is clearer why the dependency on `dragPosition`.